crates/runestick/src/vm.rs

```diff
@@ -2258,60 +2263,48 @@ impl Vm {
         Ok(())
     }
 
-    #[cfg_attr(feature = "bench", inline(never))]
-    fn op_unwrap(&mut self) -> Result<(), VmError> {
-        let value = self.stack.pop()?;
-
-        let value = match value {
-            Value::Option(option) => match &*option.borrow_ref()? {
-                Some(value) => value.clone(),
-                None => {
-                    return Err(VmError::from(VmErrorKind::UnsupportedUnwrapNone));
-                }
-            },
-            Value::Result(result) => match &*result.borrow_ref()? {
-                Ok(value) => value.clone(),
-                Err(err) => {
-                    return Err(VmError::from(VmErrorKind::UnsupportedUnwrapErr {
-                        err: err.type_info()?,
-                    }));
-                }
-            },
-            other => {
-                return Err(VmError::from(VmErrorKind::UnsupportedUnwrap {
-                    actual: other.type_info()?,
-                }));
-            }
-        };
-
-        self.stack.push(value);
-        Ok(())
-    }
-
     #[cfg_attr(feature = "bench", inline(never))]
     fn op_is_unit(&mut self) -> Result<(), VmError> {
         let value = self.stack.pop()?;
         self.stack.push(matches!(value, Value::Unit));
         Ok(())
     }
 
-    /// Test if the top of the stack is an error.
+    /// Perform the try operation on the given stack location.
     #[cfg_attr(feature = "bench", inline(never))]
-    fn op_is_value(&mut self) -> Result<(), VmError> {
-        let value = self.stack.pop()?;
+    fn op_try(
+        &mut self,
+        address: InstAddress,
+        clean: usize,
+        preserve: bool,
+    ) -> Result<bool, VmError> {
+        let value = self.stack.address_peek(address)?;
 
-        let is_value = match value {
-            Value::Result(result) => result.borrow_ref()?.is_ok(),
-            Value::Option(option) => option.borrow_ref()?.is_some(),
+        let value = match value {
+            Value::Result(result) => match &*result.borrow_ref()? {
+                Result::Ok(value) => Some(value.clone()),
+                Result::Err(..) => None,
+            },
+            Value::Option(option) => match &*option.borrow_ref()? {
+                Option::Some(value) => Some(value.clone()),
+                Option::None => None,
+            },
             other => {
                 return Err(VmError::from(VmErrorKind::UnsupportedIsValueOperand {
                     actual: other.type_info()?,
                 }))
             }
         };
 
-        self.stack.push(is_value);
-        Ok(())
+        if let Some(value) = value {
+            if preserve {
+                self.stack.push(value);
+            }
+
+            Ok(false)
```

Comment:
I'd expect that we also popped the peeked address somewhere here? Or is it because we might unwrap something that isn't top that we can't?

Reply:
Dangit. I think you're right! If `InstAddress::Top` then it should be popped according to its contract.

Reply:
Fixed now. Thank you!